hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieAvroPayload.java

```diff
@@ -40,13 +43,13 @@ public HoodieAvroPayload preCombine(HoodieAvroPayload another) {
     }
 
     @Override
-    public IndexedRecord combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema)
-        throws IOException {
+    public Optional<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema)
+            throws IOException {
         return getInsertValue(schema);
     }
 
     @Override
-    public IndexedRecord getInsertValue(Schema schema) throws IOException {
-        return HoodieAvroUtils.rewriteRecord(record, schema);
+    public Optional<IndexedRecord> getInsertValue(Schema schema) throws IOException {
+        return record.map(r -> HoodieAvroUtils.rewriteRecord(r, schema));
```

Comment:
In Hoodie Code, we have always been using Optional.of to generate an Optional of a object. It is good to change this to follow a uniform pattern